sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -856,6 +878,46 @@ def delete_port_from_status_table_hw(logical_port_name, port_mapping, status_tbl
                 continue
             status_tbl.hdel(physical_port_name, f)
 
+# Initialize the APPL_DB PROC_INFO:XCVRD table with MEDIA_NOTIFY_REQUIRED and
+# CMIS_REINIT_REQUIRED as the keys if not already populated.
+# If both the keys of these keys are set to False,
+# port_reinit_request_tbl will be set as False. In all other cases,
+# port_reinit_request_tbl will be set as True and so will be both the keys
+# Return value: port_reinit_request_tbls
+def init_appl_proc_info_tbl(namespaces, port_mapping_data, xcvr_table_helper):
+    namespace_reinit_request_tbl = {}
+    port_reinit_request_tbl = {}
+
+    for namespace in namespaces:
+        asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+        namespace_reinit_request_tbl[asic_id] = False
+        proc_info_tbl = xcvr_table_helper.get_appl_proc_info_tbl(asic_id)
+        if proc_info_tbl is None:
+            helper_logger.log_error("Appl {} tbl is None for asic_id {} during init".format(APPL_PROC_INFO_TABLE, asic_id))
+            continue
+        _, proc_info_xcvrd_fvs = proc_info_tbl.get('XCVRD')
+        if proc_info_xcvrd_fvs is not None:
+            proc_info_xcvrd_fvs_dict = dict(proc_info_xcvrd_fvs)
+            media_notify_rqd = proc_info_xcvrd_fvs_dict.get('MEDIA_NOTIFY_REQUIRED', 'True')
+            cmis_reinit_rqd = proc_info_xcvrd_fvs_dict.get('CMIS_REINIT_REQUIRED', 'True')
+            if media_notify_rqd == 'False' and cmis_reinit_rqd == 'False':
```

Comment:
Is it possible that media_notify_rqd is "True" but cmis_reinit_rqd is "False" or vice versa?

Reply:
Yes - that is a possibility if either CmisManagerTask or SfpStateUpdateTask thread crashes while performing the initialization.
In such scenario, xcvrd would perform the CMIS reinit and media setting notify for all ports in the affected namespace (irrespective of the completion status of media settings and CMIS reinit of ports before the thread got killed)